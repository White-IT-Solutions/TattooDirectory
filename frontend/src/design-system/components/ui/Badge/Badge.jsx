"use client";

import { forwardRef } from 'react';
import { cva } from '../../../utils/cn';

// Badge variant configurations using our design system
const badgeVariants = cva(
  // Base styles - common to all badges
  [
    'inline-flex items-center justify-center',
    'font-semibold transition-all duration-200 ease-out',
    'border select-none',
    'whitespace-nowrap'
  ].join(' '),
  {
    variants: {
      variant: {
        primary: [
          'text-white',
          'bg-[var(--interactive-primary)]',
          'border-[var(--interactive-primary)]',
          'hover:bg-[var(--interactive-primary-hover)]'
        ].join(' '),
        secondary: [
          'text-[var(--text-primary)]',
          'bg-[var(--interactive-secondary)]',
          'border-[var(--border-primary)]',
          'hover:bg-[var(--interactive-secondary-hover)]'
        ].join(' '),
        accent: [
          'text-white',
          'bg-[var(--interactive-accent)]',
          'border-[var(--interactive-accent)]',
          'hover:bg-[var(--interactive-accent-hover)]'
        ].join(' '),
        success: [
          'text-white',
          'bg-[var(--feedback-success)]',
          'border-[var(--feedback-success)]',
          'hover:bg-[var(--color-success-600)]'
        ].join(' '),
        warning: [
          'text-white',
          'bg-[var(--feedback-warning)]',
          'border-[var(--feedback-warning)]',
          'hover:bg-[var(--color-warning-600)]'
        ].join(' '),
        error: [
          'text-white',
          'bg-[var(--feedback-error)]',
          'border-[var(--feedback-error)]',
          'hover:bg-[var(--color-error-600)]'
        ].join(' '),
        outline: [
          'text-[var(--interactive-primary)]',
          'bg-transparent',
          'border-[var(--interactive-primary)]',
          'hover:bg-[var(--interactive-primary)] hover:text-white'
        ].join(' '),
        ghost: [
          'text-[var(--text-primary)]',
          'bg-[var(--background-secondary)]',
          'border-transparent',
          'hover:bg-[var(--interactive-secondary)]'
        ].join(' ')
      },
      size: {
        sm: [
          'h-5 px-2 text-xs',
          'rounded-[var(--radius)]',
          'gap-1'
        ].join(' '),
        md: [
          'h-6 px-2.5 text-sm',
          'rounded-[var(--radius)]',
          'gap-1.5'
        ].join(' '),
        lg: [
          'h-7 px-3 text-sm',
          'rounded-[var(--radius-md)]',
          'gap-2'
        ].join(' ')
      }
    },
    defaultVariants: {
      variant: 'primary',
      size: 'md'
    }
  }
);

// Icon component for badges
const BadgeIcon = ({ size = 'md', children, className = '' }) => {
  const iconSizes = {
    sm: 'h-3 w-3',
    md: 'h-4 w-4',
    lg: 'h-4 w-4'
  };

  return (
    <span 
      className={`${iconSizes[size]} ${className} inline-flex items-center justify-center`} 
      style={{ lineHeight: '1' }}
      aria-hidden="true"
    >
      {children}
    </span>
  );
};

const Badge = forwardRef(({ 
  className, 
  variant = 'primary', 
  size = 'md',
  icon,
  children,
  ...props 
}, ref) => {
  return (
    <span
      className={badgeVariants({ variant, size, className })}
      ref={ref}
      {...props}
    >
      {icon && (
        <BadgeIcon size={size}>
          {icon}
        </BadgeIcon>
      )}
      {children}
    </span>
  );
});

Badge.displayName = 'Badge';

export default Badge;
export { Badge, badgeVariants, BadgeIcon };