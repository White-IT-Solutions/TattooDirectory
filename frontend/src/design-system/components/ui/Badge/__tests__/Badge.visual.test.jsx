import { render } from '@testing-library/react';
import Badge from '../Badge';

describe('Badge Visual Tests', () => {
  const TestIcon = () => <span>â˜…</span>;

  it('renders all badge variants correctly', () => {
    const variants = ['primary', 'secondary', 'accent', 'success', 'warning', 'error', 'outline', 'ghost'];
    
    const { container } = render(
      <div className="flex flex-wrap gap-2 p-4">
        {variants.map(variant => (
          <Badge key={variant} variant={variant}>
            {variant}
          </Badge>
        ))}
      </div>
    );

    expect(container.firstChild).toMatchSnapshot('badge-variants');
  });

  it('renders all badge sizes correctly', () => {
    const sizes = ['sm', 'md', 'lg'];
    
    const { container } = render(
      <div className="flex items-center gap-2 p-4">
        {sizes.map(size => (
          <Badge key={size} size={size}>
            {size}
          </Badge>
        ))}
      </div>
    );

    expect(container.firstChild).toMatchSnapshot('badge-sizes');
  });

  it('renders badges with icons correctly', () => {
    const { container } = render(
      <div className="flex flex-wrap gap-2 p-4">
        <Badge icon={<TestIcon />}>With Icon</Badge>
        <Badge variant="accent" icon={<TestIcon />}>Accent Icon</Badge>
        <Badge variant="outline" size="lg" icon={<TestIcon />}>Large Outline</Badge>
      </div>
    );

    expect(container.firstChild).toMatchSnapshot('badge-with-icons');
  });

  it('renders semantic color badges correctly', () => {
    const { container } = render(
      <div className="flex flex-wrap gap-2 p-4">
        <Badge variant="success">Success</Badge>
        <Badge variant="warning">Warning</Badge>
        <Badge variant="error">Error</Badge>
      </div>
    );

    expect(container.firstChild).toMatchSnapshot('badge-semantic-colors');
  });

  it('renders badge combinations correctly', () => {
    const { container } = render(
      <div className="space-y-4 p-4">
        <div className="flex flex-wrap gap-2">
          <Badge variant="primary" size="sm">Small Primary</Badge>
          <Badge variant="secondary" size="md">Medium Secondary</Badge>
          <Badge variant="accent" size="lg">Large Accent</Badge>
        </div>
        <div className="flex flex-wrap gap-2">
          <Badge variant="outline" icon={<TestIcon />}>Outline with Icon</Badge>
          <Badge variant="ghost" icon={<TestIcon />}>Ghost with Icon</Badge>
        </div>
      </div>
    );

    expect(container.firstChild).toMatchSnapshot('badge-combinations');
  });
});