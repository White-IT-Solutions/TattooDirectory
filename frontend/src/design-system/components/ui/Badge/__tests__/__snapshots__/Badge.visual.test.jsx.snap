// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Visual Tests renders all badge sizes correctly: badge-sizes 1`] = `
<div
  class="flex items-center gap-2 p-4"
>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-5 px-2 text-xs rounded-[var(--radius)] gap-1"
  >
    sm
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    md
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-7 px-3 text-sm rounded-[var(--radius-md)] gap-2"
  >
    lg
  </span>
</div>
`;

exports[`Badge Visual Tests renders all badge variants correctly: badge-variants 1`] = `
<div
  class="flex flex-wrap gap-2 p-4"
>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    primary
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--text-primary)] bg-[var(--interactive-secondary)] border-[var(--border-primary)] hover:bg-[var(--interactive-secondary-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    secondary
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-accent)] border-[var(--interactive-accent)] hover:bg-[var(--interactive-accent-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    accent
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-success)] border-[var(--feedback-success)] hover:bg-[var(--color-success-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    success
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-warning)] border-[var(--feedback-warning)] hover:bg-[var(--color-warning-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    warning
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-error)] border-[var(--feedback-error)] hover:bg-[var(--color-error-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    error
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--interactive-primary)] bg-transparent border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary)] hover:text-white h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    outline
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--text-primary)] bg-[var(--background-secondary)] border-transparent hover:bg-[var(--interactive-secondary)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    ghost
  </span>
</div>
`;

exports[`Badge Visual Tests renders badge combinations correctly: badge-combinations 1`] = `
<div
  class="space-y-4 p-4"
>
  <div
    class="flex flex-wrap gap-2"
  >
    <span
      class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-5 px-2 text-xs rounded-[var(--radius)] gap-1"
    >
      Small Primary
    </span>
    <span
      class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--text-primary)] bg-[var(--interactive-secondary)] border-[var(--border-primary)] hover:bg-[var(--interactive-secondary-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
    >
      Medium Secondary
    </span>
    <span
      class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-accent)] border-[var(--interactive-accent)] hover:bg-[var(--interactive-accent-hover)] h-7 px-3 text-sm rounded-[var(--radius-md)] gap-2"
    >
      Large Accent
    </span>
  </div>
  <div
    class="flex flex-wrap gap-2"
  >
    <span
      class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--interactive-primary)] bg-transparent border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary)] hover:text-white h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
    >
      <span
        aria-hidden="true"
        class="h-4 w-4  inline-flex items-center justify-center"
        style="line-height: 1;"
      >
        <span>
          ★
        </span>
      </span>
      Outline with Icon
    </span>
    <span
      class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--text-primary)] bg-[var(--background-secondary)] border-transparent hover:bg-[var(--interactive-secondary)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
    >
      <span
        aria-hidden="true"
        class="h-4 w-4  inline-flex items-center justify-center"
        style="line-height: 1;"
      >
        <span>
          ★
        </span>
      </span>
      Ghost with Icon
    </span>
  </div>
</div>
`;

exports[`Badge Visual Tests renders badges with icons correctly: badge-with-icons 1`] = `
<div
  class="flex flex-wrap gap-2 p-4"
>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-primary)] border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    <span
      aria-hidden="true"
      class="h-4 w-4  inline-flex items-center justify-center"
      style="line-height: 1;"
    >
      <span>
        ★
      </span>
    </span>
    With Icon
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--interactive-accent)] border-[var(--interactive-accent)] hover:bg-[var(--interactive-accent-hover)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    <span
      aria-hidden="true"
      class="h-4 w-4  inline-flex items-center justify-center"
      style="line-height: 1;"
    >
      <span>
        ★
      </span>
    </span>
    Accent Icon
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-[var(--interactive-primary)] bg-transparent border-[var(--interactive-primary)] hover:bg-[var(--interactive-primary)] hover:text-white h-7 px-3 text-sm rounded-[var(--radius-md)] gap-2"
  >
    <span
      aria-hidden="true"
      class="h-4 w-4  inline-flex items-center justify-center"
      style="line-height: 1;"
    >
      <span>
        ★
      </span>
    </span>
    Large Outline
  </span>
</div>
`;

exports[`Badge Visual Tests renders semantic color badges correctly: badge-semantic-colors 1`] = `
<div
  class="flex flex-wrap gap-2 p-4"
>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-success)] border-[var(--feedback-success)] hover:bg-[var(--color-success-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    Success
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-warning)] border-[var(--feedback-warning)] hover:bg-[var(--color-warning-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    Warning
  </span>
  <span
    class="inline-flex items-center justify-center font-semibold transition-all duration-200 ease-out border select-none whitespace-nowrap text-white bg-[var(--feedback-error)] border-[var(--feedback-error)] hover:bg-[var(--color-error-600)] h-6 px-2.5 text-sm rounded-[var(--radius)] gap-1.5"
  >
    Error
  </span>
</div>
`;
